### PR TITLE
Lower-case column names returned from internal queries so that they work with H2

### LINKED ISSUE #32244: Some audit queries return empty results with an H2 application database
### Describe the bug

The following audit queries return empty results:
- `metabase-enterprise.audit-app.pages.dashboards/views-and-saves-by-time`
- `metabase-enterprise.audit-app.pages.users/active-and-new-by-time`

Don't forget unskip a spec in `e2e/test/scenarios/auditing/auditing.cy.spec.js` after fixing this

### To Reproduce

1. Run Metabase EE with token and using H2 as the app DB
2. Open Admin -> Audit so that it shows "Team members" page
3. "Active members and new members per day" returned no results which is not correct
4. In the sidebar click on "Dashboards"
5. "Dashboard views and saves per day" returned no results which is not correct

### Expected behavior

Charts show actual data

### Logs

_No response_

### Information about your Metabase installation

```JSON
Latest `master`, H2
```

### Severity

Breaks audit experience

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by noahmoss**
Resolves https://github.com/metabase/metabase/issues/32244

This was only an issue for H2 app DBs, because they return column names capitalized, but the Clojure post-processing code was relying on the names being lower-cased. More details in [this thread](https://metaboat.slack.com/archives/CKZEMT1MJ/p1690398803475569).

This PR fixes it by lower-casing column names and adds a basic test that repro's the issue. Audit V1 tests aren't very comprehensive (they don't hit every query) but I don't think it's worth sinking time into adding lots more since we'll ship audit v2 soon anyway.

### CODE REVIEW

File: enterprise/backend/test/metabase_enterprise/audit_app/pages_test.clj [hunk 1]
```diff
@@ -224,3 +224,15 @@
...
+
+(deftest active-and-new-by-time-test
+  (testing "The active-and-new-by-time audit query correctly returns results (#32244)"
+    (mt/with-test-user :crowberto
+      (premium-features-test/with-premium-features #{:audit-app}
+        (let [results (mt/user-http-request :crowberto :post 202 "dataset"
+                                            {:type :internal
+                                             :fn   "metabase-enterprise.audit-app.pages.users/active-and-new-by-time"
+                                             :args ["day"]})]
+          (def results results)
```
Comment (qnkhuat) [inline enterprise/backend/test/metabase_enterprise/audit_app/pages_test.clj:236]:
```suggestion
```

**Review by qnkhuat — APPROVED** [review]
LGTM but there is a stray `def`, weird that kondo didn't catch it.

**Comment by qnkhuat** [comment]
Found it, we turned off that linter for test ns . Opened this to disable that https://github.com/metabase/metabase/pull/33606

File: enterprise/backend/test/metabase_enterprise/audit_app/pages_test.clj (hunk 1 above)
Comment (noahmoss) [reply enterprise/backend/test/metabase_enterprise/audit_app/pages_test.clj:236]: thanks for catching, very weird that this didn't show up in CI

**Pull request merged**